core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/user-factory.js

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Utility File for declaring and initializing users.
+ */
+
+let e2eSuperAdmin = e2eBlogAdmin = e2eBlogPostEditor = e2eGuestUser =
+  require('../user-utilities/blog-post-admin-utils.js');
+
+/**
+ * Global user instances that are created and can be reused again.
+ */
+let superAdminInstance = null, blogAdminInstance = null,
+  blogPostEditorInstance = null;
+const ROLE_BLOG_ADMIN = 'blog admin';
+const ROLE_BLOG_POST_EDITOR = 'blog post editor';
+let browserInstances = [];
+
+/**
+ * The function creates a new super admin user and returns the instance
+ * of that user.
+ * @param {string} username - The username of the super admin.
+ * @returns The super admin instance created.
+ */
+let createNewSuperAdmin = async function(username) {
+  if (superAdminInstance !== null) {
+    return superAdminInstance;
+  }
+
+  const superAdmin = await new e2eSuperAdmin();
+  await superAdmin.openBrowser();
+  await superAdmin.signUpNewUser(username, 'testadmin@example.com');
+
+  browserInstances.push(superAdmin);
```

Comment:
Why are you pushing a "user" into a "browserInstances" array? The 'types' don't match.

Maybe activeUsers?

Reply:
yes changing this to `activeUsers`